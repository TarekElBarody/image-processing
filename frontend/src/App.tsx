import { BrowserRouter, Route, Routes } from 'react-router-dom';
import 'bootstrap/dist/css/bootstrap.min.css';

import './assets/App.css';

import Login from './pages/Login';
import Register from './pages/Register';
import Home from './pages/Home';
import Profile from './pages/Profile';
import Error404 from './pages/Error404';
import Manager from './pages/Manager';
import RegisterPage from './pages/RegisterPage';

type Props = {};

const App: React.FunctionComponent<Props> = (props) => {
    return (
        //const { currentUser } = this.state;

        <BrowserRouter>
            <Routes>
                <Route path="/" element={<Home />} />
                <Route path="/home" element={<Home />} />
                <Route path="/register-page" element={<RegisterPage />} />
                <Route path="/login" element={<Login history="" />} />
                <Route path="/register" element={<Register />} />
                <Route path="/profile" element={<Profile />} />
                <Route path="/manager" element={<Manager />} />

                <Route path="*" element={<Error404 />} />
            </Routes>
        </BrowserRouter>
    );
};

export default App;
