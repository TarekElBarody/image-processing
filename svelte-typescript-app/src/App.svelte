<script>
	import { Router, Route, Link } from "svelte-navigator";
	import Login from "./routes/login.svelte";
	import Home from "./routes/home.svelte";
	import TopNave from "./components/topNav.svelte";
  import { useNavigate, useLocation } from "svelte-navigator";
	import { user, isAuthenticated, token } from "./lib/store";

	import PrivateRoute from "./routes/PrivateRoute.svelte";
	import { onMount } from "svelte";
  
	let auth = false;
	isAuthenticated.subscribe(a=> auth = a);

	function handleLogout() {
		$token='';
	 $isAuthenticated = false;
        $user = {
			id: null,
			first_name: null,
			last_name: null,
			role: null
		};
    	localStorage.setItem("token", $token);
	};

	
	  
	
  </script>
  
  <Router> 
	<main class="wrap">
		{#if auth === false}
		  <Route path="login">
		<Login />
	  </Route>
	  {:else}
	  <TopNave/>
		{/if}
	
  
	  <PrivateRoute path="/">
		<Home/>
	  </PrivateRoute>

	  <PrivateRoute path="/home">
		<Home/>
	  </PrivateRoute>
  
	  <PrivateRoute path="/about">
		<h3>About</h3>
		<p>That's what it's all about!</p>
	  </PrivateRoute>
  
	  <PrivateRoute path="/profile" let:location>
		<h3>Welcome {$user.first_name}</h3>
		<button on:click={handleLogout}>Logout</button>
	  </PrivateRoute>

	  <Route>
		  <h3>404 Not Found!</h3>
	  </Route>
	</main>
  </Router>
  
  {#if auth == true}
		<style>
			body {
				background: #f5f5f5;
			}
		</style>
		 
	  {:else}
	  <style>
		body{
			background: #fff;
		}
	</style>
		{/if}

  <style>
	  .wrap{width: 100%;}
  </style>