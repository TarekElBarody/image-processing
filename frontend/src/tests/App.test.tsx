/* eslint-disable testing-library/no-unnecessary-act */
import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, fireEvent, render, screen, act } from '@testing-library/react';

import App from '../App';
afterEach(cleanup);

test('App', async () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON() as renderer.ReactTestRendererJSON;
    expect(tree).toMatchSnapshot();
    render(<App />);
    // manually trigger the callback
    const button = screen.getByRole('button', { name: 'Login' });
    expect(button).toBeInTheDocument();
    const email = screen.getByRole('textbox', { name: 'Email address' });
    expect(email).toBeInTheDocument();
    const password = screen.getByLabelText('Password');
    expect(password).toBeInTheDocument();
    await act(async () => {
        fireEvent.change(email, { target: { value: 'admin@admin.com' } });
        fireEvent.change(password, { target: { value: '123456789' } });
        fireEvent.click(button);
    });
    // manually trigger the callback
    //tree.children?.find('form')?.children?.find('email');
    expect(screen.getByText('Process Logging in ..')).toBeInTheDocument();

    // re-rendering
    tree = component.toJSON() as renderer.ReactTestRendererJSON;
    expect(tree).toMatchSnapshot();
});
