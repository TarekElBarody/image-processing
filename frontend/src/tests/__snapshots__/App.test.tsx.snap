// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App 1`] = `
<main
  className="form-login"
>
  <h3
    className="h3 mb-3 fw-normal"
  >
    Please sign in
  </h3>
  <hr />
  <form
    action="#"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="form-floating"
    >
      <div
        className="form-group mb-3"
      >
        <label
          htmlFor="email"
        >
          Email address
        </label>
        <input
          autoComplete="email"
          className="form-control"
          id="email"
          label="email"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        />
        <p
          className="text-muted"
        >
          We'll never share your email with anyone else.
        </p>
      </div>
    </div>
    <div
      className="form-floating"
    >
      <div
        className="form-group mb-3"
      >
        <label
          htmlFor="password"
        >
          Password
        </label>
        <input
          autoComplete="current-password"
          className="form-control"
          id="password"
          label="password"
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      className="form-group"
    />
    <button
      className="w-100 btn btn-lg btn-primary"
      name="Login"
      type="submit"
    >
      <span>
        Login
      </span>
    </button>
  </form>
  <hr
    className="mt"
  />
  <p
    className="mt-1 mb-1 center"
  >
    If you dont have account .. 
    <a
      href="/register"
    >
      Create New
    </a>
  </p>
  <p
    className="mt-5 mb-1 center"
  >
    <a
      href="https://github.com/TarekElBarody"
    >
      Tarek El-Barody
    </a>
     © 2022
  </p>
</main>
`;

exports[`App 2`] = `
<main
  className="form-login"
>
  <h3
    className="h3 mb-3 fw-normal"
  >
    Please sign in
  </h3>
  <hr />
  <form
    action="#"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="form-floating"
    >
      <div
        className="form-group mb-3"
      >
        <label
          htmlFor="email"
        >
          Email address
        </label>
        <input
          autoComplete="email"
          className="form-control"
          id="email"
          label="email"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        />
        <p
          className="text-muted"
        >
          We'll never share your email with anyone else.
        </p>
      </div>
    </div>
    <div
      className="form-floating"
    >
      <div
        className="form-group mb-3"
      >
        <label
          htmlFor="password"
        >
          Password
        </label>
        <input
          autoComplete="current-password"
          className="form-control"
          id="password"
          label="password"
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      className="form-group"
    />
    <button
      className="w-100 btn btn-lg btn-primary"
      name="Login"
      type="submit"
    >
      <span>
        Login
      </span>
    </button>
  </form>
  <hr
    className="mt"
  />
  <p
    className="mt-1 mb-1 center"
  >
    If you dont have account .. 
    <a
      href="/register"
    >
      Create New
    </a>
  </p>
  <p
    className="mt-5 mb-1 center"
  >
    <a
      href="https://github.com/TarekElBarody"
    >
      Tarek El-Barody
    </a>
     © 2022
  </p>
</main>
`;
